scene/main/node.cpp

```diff
@@ -1315,7 +1315,7 @@ void Node::_generate_serial_child_name(const Node *p_child, StringName &name) co
 	if (name == StringName()) {
 		// No name and a new name is needed, create one.
 
-		name = p_child->get_class();
+		name = adjust_name_casing(p_child->get_class());
```

Comment:
Hold on here. This also adjusts the name when `force_readable_name` is true in Node's `add_child`.

It's worth keeping in mind if this is desired (in my opinion, yes).